pkg/controller/external.go

```diff
@@ -291,3 +284,39 @@ func CriticalAnnotationsCallback(kube client.Client, tr resource.Terraformed) te
 		return errors.Wrap(c.UpdateCriticalAnnotations(ctx, tr), "cannot update critical annotations")
 	}
 }
+
+func getConnectionDetails(attr map[string]interface{}, tr resource.Terraformed, cfg config.Resource) (managed.ConnectionDetails, error) {
```

Comment:
I'm not sure if we should have `getConnectionDetails` and `lateInitializeAnnotations` functions here. They seem like resource operations, hence could go to `resource` package. What do you think?